src/common/libutil/stdlog.c

```diff
@@ -141,6 +141,37 @@ int stdlog_decode (const char *buf, int len, struct stdlog_header *hdr,
     return 0;
 }
 
+char *stdlog_split_message (const char *buf, int *len, const char *sep)
+{
+    struct stdlog_header hdr;
+    const char *msg;
+    int msglen;
+    int off;
+    char *xtra;
+    int xtra_len;
+
+    if (stdlog_decode (buf, *len, &hdr, NULL, NULL, &msg, &msglen) < 0)
+        return NULL;
+    for (off = 0; off < msglen; off++) {
+        if (strchr (sep, msg[off]))
+            break;
+    }
+    xtra_len = msglen - off;
+    if (xtra_len == 0)
+        return NULL;
+    if (!(xtra = malloc (xtra_len + 1)))
+        return NULL;
+    memcpy (xtra, msg + off, xtra_len);
+    xtra[xtra_len] = '\0';
+    *len -= xtra_len; // truncate original (N.B. message is last, no \0 term)
+    while (xtra_len > 0 && strchr (sep, xtra[0]))
+        memmove (xtra, xtra + 1, xtra_len--); // drop leading separator(s)
```

Comment:
Just curious if this `memmove` could be eliminated if the offset was advanced past any leading separator(s) before `memcpy`?

Reply:
Yep that would have been a little better, though I'm inclined to agree that changing it now might not be worth the effort given that we don't expect multi-line log messages to occur much.

Reply:
yeah, perfectly fine, I was more curious if there was something I was missing..